### PR TITLE
Make squids great again

### PR DESCRIPTION
**Pull request by Minepolz320**
Squids make sounds and shoot inks

### CODE REVIEW

File: src/main/java/jss/bugtorch/features/squidFix/FixSquidSound.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import java.util.Random;
+
+
+
+
+public class FixSquidSound {
+
+
+    boolean isPass(LivingEvent event) {
+        return !event.entity.worldObj.isRemote && EntitySquid.class.equals(event.entity.getClass());
```
Comment (Alexdoru) [inline src/main/java/jss/bugtorch/features/squidFix/FixSquidSound.java]: event.entity instanceof EntitySquid

Comment (Minepolz320) [reply src/main/java/jss/bugtorch/features/squidFix/FixSquidSound.java]: @Alexdoru Yep but i want add this only for vanilla squids

**Comment by Alexdoru** [comment]
instead of subscribing to events, could you use mixins and @Overwrite these methods in EntitySquid 
image

**Comment by Minepolz320** [comment]
Unfortunately, I'm not very good at using mixins.
I feel forge be more stable for me

**Comment by Alexdoru** [comment]
look in the same project at other mixins that use the @Overwrite annotation. and then register your mixin here https://github.com/GTNewHorizons/BugTorch/blob/7fb2459a7d3afc97468c551a0e4eb881c75a8905/src/main/java/jss/bugtorch/core/BugTorchMixinPlugin.java#L120

**Comment by Glease** [comment]
shouldn't event be preferred over mixin, especially an `@Overwrite`?

**Comment by mitchej123** [comment]
Events seem fine in this case.

**Comment by glowredman** [comment]
Where are the sound files from? @Minepolz320 did you create them yourself?

**Comment by Alexdoru** [comment]
> Events seem fine in this case.

shouldn't we try to minimise the use of @SubscribeEvent to minimise the amount of code running all the time

**Comment by mitchej123** [comment]
> shouldn't we try to minimise the use of @SubscribeEvent to minimise the amount of code running all the time

Mixins/ASM cause different issues; they're a great tool but not a silver bullet.  One additional event listener isn't going to cause much overhead, especially if it has a low cost early out.

**Comment by glowredman** [comment]
Generally one should always look if there are other solutions before using ASM/Mixins.

**Comment by Alexdoru** [comment]
but in this case the calls to play sounds are already in EntityLiving and you can add sounds by just making the methods getLivingSound() ... return something. And before doing ASM/mixins one should always look at the full pack class dump to avoid compatibility issues

**Comment by eigenraven** [comment]
You need to run the `spotlessApply` gradle task on this

**Comment by RinWate** [comment]
I think using byte code modification is not recommended i think. Byte code modification can be made a lot of errors in other parts of code.
P.S. ASM /mixins is only need to be used in very specific situations.

**Comment by Alexdoru** [comment]
why everyone so afraid of mixins

**Comment by Minepolz320** [comment]
UP?

**Comment by boubou19** [comment]
> UP?

Well test failed, so PR needs adjustment

**Comment by Minepolz320** [comment]
@boubou19 something wrong with code? 
and what is "spotlessApply" anyway

**Comment by boubou19** [comment]
you need to run `./gradlew SpotlessApply`, it's our linter. It ensures we all have the same formatting in code to avoid weird stuff.

**Comment by Alexdoru** [comment]
did you tell us where are the audio files from

**Comment by Minepolz320** [comment]
@Alexdoru from Minecraft wiki
 it can cause serious problems?

**Comment by mitchej123** [comment]
> @Alexdoru from Minecraft wiki it can cause serious problems?

Potential licensing issues.  What license is listed for the files?  Got a link?

**Comment by Minepolz320** [comment]
@mitchej123 https://minecraft.fandom.com/wiki/Squid

**Comment by Alexdoru** [comment]
> @mitchej123 https://minecraft.fandom.com/wiki/Squid

So it's assests from Mojang ? ...

**Comment by Minepolz320** [comment]
@Alexdoru even so, this forge modloader thing already uses Minecraft Mojang asset's, what about sounds from my logic if i take sounds from community Wiki it means someone uploaded it on page without any copyright issues, plus this project already contains sounds from Mojang 


TLDR: Sorry guys for me this copyright things very new for me 😅

"Community content is available under CC BY-NC-SA 3.0 unless otherwise noted."

**Comment by Minepolz320** [comment]
Why test failed again? 🤔
it builds for me...

**Comment by Alexdoru** [comment]
> @Alexdoru even so, this forge modloader thing already uses Minecraft Mojang asset's, what about sounds from my logic if i take sounds from community Wiki it means someone uploaded it on page without any copyright issues, plus this project already contains sounds from Mojang 
> 
> 
> 
> 
> 
> TLDR: Sorry guys for me this copyright things very new for me 😅
> 
> 
> 
> "Community content is available under CC BY-NC-SA 3.0 unless otherwise noted."
> 
> 

Forge mod loader doesn't distribute assets from mojang, forge mod loader injects its code into the vanilla minecraft code

**Comment by Minepolz320** [comment]
@Alexdoru ok, thanks for explanation

**Comment by leagris** [comment]
@Minepolz320 Your editor or IDE must be configured to Unix `LF` line-ending or use the provided `.editorconfig`.

Anyway, to never forget the `:spotlessapply` task before committing; I wrote this Git pre-commit hook
`.git/hooks/pre-commit`:
```sh
#!/bin/sh
# pre-commit hook for GTNH projects

# Blocks commits to master branch
branch_name="$(git rev-parse --abbrev-ref HEAD)"
if [ master = "${branch_name}" ]; then
  printf 'Commit to %s branch denied!\n' "${branch_name}" >&2
  exit 1 # Reject commit
fi

# Forces spotless on new commit's actionable files
./gradlew :spotlessApply
```

You may be able to adapt it for the Windows environment:

**Comment by Dream-Master** [comment]
@Minepolz320 you need to change imports using new gtnh misin

**Comment by Dream-Master** [comment]
image

**Comment by glowredman** [comment]
> Where are the sound files from? @Minepolz320 did you create them yourself?

**Comment by minecraft7771** [comment]
> > Where are the sound files from? @Minepolz320 did you create them yourself?

A few comments above he wrote they are taken from minecraft wiki

**Comment by Minepolz320** [comment]
@glowredman

**Comment by glowredman** [comment]
Okay, so what license are they under? If they took them from the game, we are not allowed to use them (per Mojang's TOS)

**Comment by Minepolz320** [comment]
> Freely licensed or public domain images are strongly preferred on Fandom.
Non-text media on Fandom should not be assumed to be available under the same license as the text. Please view the media description page for details about the license of any specific media file.

> If you are uploading files to Fandom, you should cite the source of the file, attribute the authors, and note any [copyright information](https://community.fandom.com/wiki/Help:File_copyright_tags), where applicable. Individual communities may elaborate upon and refine requirements for file uploads.
> Please note that all contributions to the Minecraft Wiki are considered to be released under the CC BY-NC-SA 3.0

I think even if something goes wrong we just remove sounds form here

**Comment by glowredman** [comment]
It doesn't work like that, we can't just say "yeah we read that license but we chose to ignore it until something happens"

**Comment by Minepolz320** [comment]
@glowredman soo how we can solve this issue? 
Looks like here no copyright issues if so check https://minecraft.fandom.com/wiki/Music_Disc 
>Tracks by [C418](https://minecraft.fandom.com/wiki/C418) have been shortened to 30 seconds on this wiki, due to an agreement with the composer. "11" is exempt from this.

**Comment by Minepolz320** [comment]
@Dream-Master part that uses mixin doesn't work after changing imports
or is it because I'm running from development environment

**Comment by Dream-Master** [comment]
> @Dream-Master part that uses mixin doesn't work after changing imports or is it because I'm running from development environment

idk but please update the mixin we use (GTNH mixin)

**Comment by Minepolz320** [comment]
Ok Let's try

**Comment by Dream-Master** [comment]
works for me ingame

**Review by minecraft7771 — APPROVED** [review]
